xdem/spatial_tools.py

```diff
@@ -78,3 +86,77 @@ def subtract_rasters(first_raster: Union[str, gu.georaster.Raster], second_raste
     )
 
     return difference_raster
+
+
+def merge_bounding_boxes(bounds: list[rio.coords.BoundingBox], resolution: float) -> rio.coords.BoundingBox:
+    max_bounds = dict(zip(["left", "right", "top", "bottom"], [np.nan] * 4))
+    for bound in bounds:
+        for key in "right", "top":
+            max_bounds[key] = np.nanmax([max_bounds[key], bound.__getattribute__(key)])
+        for key in "bottom", "left":
+            max_bounds[key] = np.nanmin([max_bounds[key], bound.__getattribute__(key)])
+
+    for key in max_bounds:
+        modulo = max_bounds[key] % resolution
+        max_bounds[key] -= modulo
+
+        if key in ["right", "top"] and modulo > 0:
+            max_bounds[key] += resolution
+
+    return rio.coords.BoundingBox(**max_bounds)
+
+
+def merge_rasters(rasters: list[gu.georaster.Raster], reference: int = 0, merge_algorithm: Callable = np.nanmean,
```

Comment:
Amaury will argue here that this type of method belongs in geoutils ;) (and I do too)

Reply:
> Looks good. I'm thinking of how much we should link Dask to numpy/xarray for those calculations (speed/chunking). In any case, this should be easy enough to implement at a later stage "improvement" if we feel the need to

I totally agree that this would be a great addition! This will work for small to medium sized rasters, but not necessarily for huge ones.

I feel like we should have a test framework for huge DEMs, which would highlight where performance improvements should be made. I can make an associated issue!

Reply:
@rhugonnet  I agree that it would be a great addition to geoutils. Maybe we can remove it later if it's added there?

Reply:
Please add to the list of geoutils issues and then I can add it there!